### ISSUE TITLE
Unable to install ufal.udpipe 

### ISSUE THREAD

**Issue by sslavian812** [ufal/udpipe#102]
I'm trying to install your library in Ubuntu, but it fails to compile during installation.
Do you maybe have some idea of what should I look at?

```sh
$ pip3 install ufal.udpipe==1.2.0.2
Collecting ufal.udpipe==1.2.0.2
  Using cached https://files.pythonhosted.org/packages/99/1a/9fbf9658cb3a24bfa2e462b4fda554f48b6fe7a7ee44f5036b1f7d181ca8/ufal.udpipe-1.2.0.2.tar.gz
Building wheels for collected packages: ufal.udpipe
  Running setup.py bdist_wheel for ufal.udpipe ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-giydfcbf/ufal.udpipe/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpbmshrmpwpip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/ufal
  copying ufal/__init__.py -> build/lib.linux-x86_64-3.6/ufal
  copying ufal/udpipe.py -> build/lib.linux-x86_64-3.6/ufal
  running build_ext
  building 'ufal_udpipe' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/udpipe
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iudpipe/include -I/usr/include/python3.6m -c udpipe/udpipe.cpp -o build/temp.linux-x86_64-3.6/udpipe/udpipe.o -std=c++11 -fvisibility=hidden -w
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iudpipe/include -I/usr/include/python3.6m -c udpipe/udpipe_python.cpp -o build/temp.linux-x86_64-3.6/udpipe/udpipe_python.o -std=c++11 -fvisibility=hidden -w
  udpipe/udpipe_python.cpp:184:11: fatal error: Python.h: No such file or directory
   # include <Python.h>
             ^~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for ufal.udpipe
  Running setup.py clean for ufal.udpipe
Failed to build ufal.udpipe
Installing collected packages: ufal.udpipe
  Running setup.py install for ufal.udpipe ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-giydfcbf/ufal.udpipe/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-kovar4ai-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/ufal
    copying ufal/__init__.py -> build/lib.linux-x86_64-3.6/ufal
    copying ufal/udpipe.py -> build/lib.linux-x86_64-3.6/ufal
    running build_ext
    building 'ufal_udpipe' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/udpipe
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iudpipe/include -I/usr/include/python3.6m -c udpipe/udpipe.cpp -o build/temp.linux-x86_64-3.6/udpipe/udpipe.o -std=c++11 -fvisibility=hidden -w
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iudpipe/include -I/usr/include/python3.6m -c udpipe/udpipe_python.cpp -o build/temp.linux-x86_64-3.6/udpipe/udpipe_python.o -std=c++11 -fvisibility=hidden -w
    udpipe/udpipe_python.cpp:184:11: fatal error: Python.h: No such file or directory
     # include <Python.h>
               ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-giydfcbf/ufal.udpipe/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-kovar4ai-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-giydfcbf/ufal.udpipe/
```

```sh
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
```

```sh
$ pip3 --version
pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)
```

**Comment by foxik** [comment]
You currently need also a C++11 compiler (`g++`) and Python development library (`libpython-dev`). We are planning to add support for Python wheels (so that the binaries would be precompiled and you would only download it), but it is not our top priority.

**Comment by foxik** [comment]
Closing; feel free to reopen if needed.

**Comment by plefebvre91** [comment]
I've some trouble to install ufal.udpipe python package in a Docker container. It seems it tries to compile wheel file (Python 3 / Ubuntu 20.04). 

You said you were planning to add wheels support, do these files exist now ?

**Comment by foxik** [comment]
The latest wheels are only for Windows and OS X; the Linux ones should be added soon (but with the current situation I have little time, so it may take several weeks).

**Comment by EmilStenstrom** [comment]
Just wanted to add that I'm also interested in wheels for Linux. I currently install the ufal.udpipe package on a really low powered VPS (to save money) and building the wheel takes ~15 minutes (at least). Speeding this up would we much appreciated.

**Comment by plefebvre91** [comment]
I gave up and choose to use udpipe binary through a subprocess. If it can help [code is here](https://gist.github.com/plefebvre91/ed29f837e5a14d0e6817889845acf33a)

**Comment by foxik** [comment]
Unfortunately, with the pandemic I had to stop working on UDPipe and other software, so this is still not done. But the wheels will appear at some point...

**Comment by EmilStenstrom** [comment]
@foxik Didn't mean to put pressure on you! Wanted to add a vote to help with prioritization (I have no idea what other things are planned), when you do have a chance to continue working on udpipe.

**Comment by EmilStenstrom** [comment]
@plefebvre91 Happy to hear you use the conllu library, I'm the author of it ;)

**Comment by yuoppp** [comment]
So, It's 2023. Any updates on this issue? Maybe someone found a workaround on it or forked the library?

**Comment by martinpopel** [comment]
See #165

**Comment by foxik** [comment]
Hi @youppp, the binary wheels for UDPipe 1.2.0.3 have been available for two days now; they cover Python 3.6-3.11, Intel 32bit, 64bit for Windows and Linux, and 64bit Intel and Arm for macOS. If they are not working for you, please create a separate issue. Cheers!
